### PR TITLE
Account Hook and Wire up account index page

### PR DESCRIPTION
**Pull request by Benzbeeb**

### CODE REVIEW

File: scan/src/pages/AccountIndexPage.re [hunk 1]
```diff
@@ -143,19 +165,51 @@ let make = (~address, ~hashtag: Route.account_tab_t) => {
...
+               {totalBalance(
+                  "TOTAL BAND IN USD ($3.42 / BAND)",
+                  balance *. price |> Format.fPretty,
+                  "USD",
+                )}
+             </>
+           | _ =>
+             <>
+               {totalBalance("TOTAL BAND BALANCE", "?", "BAND")}
+               {totalBalance("TOTAL BAND IN USD ($3.42 / BAND)", "?", "USD")}
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: Remove price mock

File: scan/src/pages/AccountIndexPage.re [hunk 2]
```diff
@@ -118,6 +118,28 @@ let totalBalance = (title, amount, symbol) => {
...
+    |> List.map(((validatorAddress, amount, reward)) =>
+         (validatorAddress |> Address.toBech32, amount, reward)
+       );
+
+  let usdPrice = {
+    let%Opt price = priceOpt;
+    Some(price.usdPrice);
+  };
+
+  let avialableBalance = {
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: Change to totalBalance = balance + stake + reward

File: scan/src/rpc/AccountHook.re [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  let addressStr = address |> Address.toBech32;
+  let json = AxiosHooks.use({j|distribution/delegators/$addressStr/rewards|j});
+
+  let balances = json |> Belt.Option.map(_, decodeStakes);
+  switch (balances) {
+  | None => None
+  | Some(x) => Some(x |> Belt_List.reduce(_, 0.0, (x, y) => x +. y))
+  };
+};
+
```
Comment (taobun) [inline scan/src/rpc/AccountHook.re:24]: Add getReward from this [endpoint](http://d3n.bandprotocol.com/distribution/delegators/band1p40yh3zkmhcv0ecqp3mcazy83sa57rgjp07dun/rewards)

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/AccountIndexPage.re [hunk 4]
```diff
@@ -143,19 +160,60 @@ let make = (~address, ~hashtag: Route.account_tab_t) => {
...
+               {totalBalance(
+                  "TOTAL BAND IN USD ($3.42 / BAND)",
+                  balance *. price |> Format.fPretty,
+                  "USD",
+                )}
+             </>
+           | _ =>
+             <>
+               {totalBalance("TOTAL BAND BALANCE", "?", "BAND")}
+               {totalBalance("TOTAL BAND IN USD ($3.42 / BAND)", "?", "USD")}
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: 3.42 !!!

File: scan/src/rpc/AccountHook.re [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    validatorAddress: string,
+    balance: float,
+  };
+
+  type reward_t = {
+    validatorAddress: string,
+    reward: float,
+  };
+
+  type merge_t = {
```
Comment (taobun) [inline scan/src/rpc/AccountHook.re]: Rename to type t

File: scan/src/rpc/AccountHook.re [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+let getBalance = address => {
+  let addressStr = address |> Address.toBech32;
+  let json = AxiosHooks.use({j|bank/balances/$addressStr|j});
+
+  let balances = json |> Belt.Option.map(_, decodeBalances);
+
+  switch (balances) {
+  | None => None
+  | Some(x) => Some(x |> Belt_List.reduce(_, 0.0, (+.)))
```
Comment (taobun) [inline scan/src/rpc/AccountHook.re]: Get only band amount not sum of all coins

File: scan/src/rpc/AccountHook.re [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+      json |> field("validator_address", string),
+      json
+      |> field("reward", list(decodeRewardUband))
+      |> Belt_List.reduce(_, 0.0, (_, (x, y)) => +. (compare("uband", y) == 0 ? x : 0.)),
+    );
+  let decodeRewards = json =>
+    JsonUtils.Decode.(json |> at(["result", "rewards"], list(decodeReward)));
+};
+
+let decodeDelegation = (delegationJson, rewardJson) =>
```
Comment (taobun) [inline scan/src/rpc/AccountHook.re]: Where this code is used?

File: scan/src/rpc/AccountHook.re [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  JsonUtils.Decode.(
+    delegationJson |> field("validator", string),
+    rewardJson |> field("amount", uamount),
+  );
+
+let decodeDelegations = (delegationJsons, rewardJsons) =>
```
Comment (taobun) [inline scan/src/rpc/AccountHook.re]: ditto

File: scan/src/pages/AccountIndexPage.re [hunk 9]
```diff
@@ -143,19 +160,60 @@ let make = (~address, ~hashtag: Route.account_tab_t) => {
...
       </Col>
       <div className=Styles.separatorLine />
       <Col size=1. alignSelf=Col.Start>
         <div className=Styles.totalContainer>
-          {totalBalance("TOTAL BAND BALANCE", "12,408,746.93", "BAND")}
-          {totalBalance("TOTAL BAND IN USD ($3.42 / BAND)", "37,226,240.79", "USD")}
+          {switch (balanceOpt, usdPrice) {
+           | (Some(balance), Some(price)) =>
+             <>
+               {totalBalance("TOTAL BAND BALANCE", balance |> Format.fPretty, "BAND")}
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: The balance must be sum of 3 values.

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/AccountIndexPage.re [hunk 10]
```diff
@@ -118,6 +118,24 @@ let totalBalance = (title, amount, symbol) => {
...
+  let priceOpt = PriceHook.get();
+  let delegations =
+    {
+      let%Opt account = accountOpt;
+      Some(account.delegations);
+    }
+    |> Belt_Option.getWithDefault(_, []);
+  let usdPrice = {
+    let%Opt price = priceOpt;
+    Some(1. /. price.usdPrice);
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: Why we need to use bandPrice = 1/bandPrice. Current usd price is 0.3 not 3.2?

File: scan/src/pages/AccountIndexPage.re [hunk 11]
```diff
@@ -118,6 +118,24 @@ let totalBalance = (title, amount, symbol) => {
...
+  };
+
+  let totalBalanceOpt = {
+    let%Opt account = accountOpt;
+    Some(account.balance +. account.balanceStake);
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: Add reward too

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/images/Images.re [hunk 12]
```diff
@@ -37,3 +37,4 @@
 [@bs.module] external leftAngle: string = "./left-angle.svg";
 [@bs.module] external graphBG: string = "./graph-bg.svg";
 [@bs.module] external notFoundBg: string = "./notFoundBg.svg";
+[@bs.module] external txLogo: string = "./tx-icon.png";
```
Comment (taobun) [inline scan/src/images/Images.re]: Already declare?

File: scan/src/contexts/GlobalContext.re [hunk 13]
```diff
@@ -12,6 +12,7 @@ let make = (~children) => {
   let financialOpt = PriceHook.get();
   let latestBlocksOpt = BlockHook.latest();
   let validatorsOpt = ValidatorHook.getList();
+
```
Comment (taobun) [inline scan/src/contexts/GlobalContext.re]: ?

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/images/Images.re (hunk 12 above)
Comment (Benzbeeb) [reply scan/src/images/Images.re]: it's a transparent background, maybe I will change the name for not miss communication?

**Comment by evilpeach** [comment]
@Benzbeeb test failed because of `tx-icon.png`. Please check.

**Review by taobun — DISMISSED** [review]

File: scan/src/components/account/AccountIndexDelegations.re [hunk 14]
```diff
@@ -79,7 +75,7 @@ let make = () => {
                      block=true
                    />
                    <Text
-                     value={validator->Js.String.sliceToEnd(~from=11)}
+                     value={delegation.validatorAddress->Js.String.sliceToEnd(~from=11)}
                      spacing={Text.Em(0.02)}
                      block=true
                      code=true
```
Comment (evilpeach) [inline scan/src/components/account/AccountIndexDelegations.re]: Let's use ` <AddressRender address validator=true />`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/pages/AccountIndexPage.re [hunk 15]
```diff
@@ -143,19 +157,60 @@ let make = (~address, ~hashtag: Route.account_tab_t) => {
...
-        {balanceDetail("AVAILABLE BALANCE", "10,547,434.89", "4,829,360.21", "5269FF")}
+        {switch (accountOpt, priceOpt) {
+         | (Some(account), Some(price)) =>
+           balanceDetail(
+             "AVAILABLE BALANCE",
+             account.balance |> Format.fPretty,
+             account.balance *. price.usdPrice |> Format.fPretty,
+             "5269FF",
+           )
+         | _ => balanceDetail("AVAILABLE BALANCE", "?", "?", "5269FF")
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: Using Color.bandBlue

File: scan/src/pages/AccountIndexPage.re [hunk 16]
```diff
@@ -143,19 +157,60 @@ let make = (~address, ~hashtag: Route.account_tab_t) => {
...
-        {balanceDetail("BALANCE AT STAKE", "1,800,000.00", "158,303.88", "ABB6FF")}
+        {switch (accountOpt, priceOpt) {
+         | (Some(account), Some(price)) =>
+           balanceDetail(
+             "BALANCE AT STAKE",
+             account.balanceStake |> Format.fPretty,
+             account.balanceStake *. price.usdPrice |> Format.fPretty,
+             "ABB6FF",
+           )
+         | _ => balanceDetail("BALANCE AT STAKE", "?", "?", "ABB6FF")
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: No have color code in Color.RE

File: scan/src/pages/AccountIndexPage.re [hunk 17]
```diff
@@ -143,19 +157,60 @@ let make = (~address, ~hashtag: Route.account_tab_t) => {
...
         <VSpacing size=Spacing.xl />
         <VSpacing size=Spacing.md />
-        {balanceDetail("REWARD", "61,301.04", "60.31", "000C5C")}
+        {switch (accountOpt, priceOpt) {
+         | (Some(account), Some(price)) =>
+           balanceDetail(
+             "REWARD",
+             account.reward |> Format.fPretty,
+             account.reward *. price.usdPrice |> Format.fPretty,
+             "000C5C",
```
Comment (taobun) [inline scan/src/pages/AccountIndexPage.re]: Color

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by prin-r — APPROVED** [review]

File: scan/src/Colors.re [hunk 18]
```diff
@@ -61,6 +61,8 @@ let blue5 = Css.hex("475DEB");
 let blue6 = Css.hex("243CDB");
 let blue7 = Css.hex("142AB8");
 let blue8 = Css.hex("0B1D8E");
+let blue9 = Css.hex("ABB6FF");
+let blue10 = Css.hex("000C5C");
```
Comment (prin-r) [inline scan/src/Colors.re]: Sorry my bad, I did't tell you that colors are sorted by its lightness. So in this case blue9 should be darker than blue8.
I think we just seperate these char colors to their own place.

let chartBalanceAtStake = Css.hex("ABB6FF");
let chartReward = Css.hex("000C5C");

Please append these lines to the bottom of Colors.re

**Review by prin-r — CHANGES_REQUESTED** [review]
Sorry my bad, I did't tell you that colors are sorted by its lightness. So in this case blue9 should be darker than blue8.
I think we just seperate these char colors to their own place.

let chartBalanceAtStake = Css.hex("ABB6FF");
let chartReward = Css.hex("000C5C");

Please append these lines to the bottom of Colors.re

**Review by prin-r — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**